lightningd/options.c

```diff
@@ -191,6 +191,27 @@ static char *opt_add_addr(const char *arg, struct lightningd *ld)
 	return opt_add_addr_withtype(arg, ld, ADDR_LISTEN_AND_ANNOUNCE, true);
 }
 
+static char *opt_alt_subdaemon(const char *arg, struct lightningd *ld)
+{
+	/* example arg: "lightning_hsmd:remote_hsmd" */
+	char *argbuf = tal_strdup(ld, arg);
+	char *colon = strchr(argbuf, ':');
+	if (!colon)
+		return tal_fmt(NULL, "argument must contain \':\'");
+	*colon = '\0';
+
+	if (!is_subdaemon(argbuf))
+		return tal_fmt(NULL, "%s is not a subdaemon", argbuf);
+
+	char *subdaemon = tal_strdup(ld, argbuf);
+	char *sdpath = tal_strdup(ld, colon+1);
+	if (!strmap_add(&ld->alt_subdaemons, subdaemon, sdpath))
+		return tal_fmt(NULL, "%s already exists", argbuf);
+
+	tal_free(argbuf);
+	return NULL;
```

Comment:
I think tal_strsplit will do this slightly more neatly.

And in general, later args should replace earlier ones. So free the old one if any, rather than complain?

Reply:
done